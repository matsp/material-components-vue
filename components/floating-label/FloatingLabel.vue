<template>
  <label
    :class="classes"
    class="mdc-floating-label"
    v-bind="$attrs"
    @_init="onParentInit"
  >
    <slot />
  </label>
</template>

<script>
import { MDCFloatingLabel } from '@material/floating-label'

import { baseComponentMixin, themeClassMixin } from '../base'

export default {
  mixins: [baseComponentMixin, themeClassMixin],
  inject: {
    getLabel: {
      default: null
    }
  },
  props: {
    floatAbove: {
      type: Boolean,
      default: false
    },
    shake: {
      type: Boolean,
      default: false
    }
  },
  data () {
    return {
      mdcFloatingLabel: undefined
    }
  },
  computed: {
    classes () {
      return {
        'mdc-floating-label--float-above': this.floatAbove,
        'mdc-floating-label--shake': this.shake
      }
    }
  },
  mounted () {
    if (!(this.getLabel instanceof Function)) { // can not be init by parent
      this.mdcFloatingLabel = MDCFloatingLabel.attachTo(this.$el)
    }
  },
  beforeDestroy () {
    if (this.mdcFloatingLabel instanceof MDCFloatingLabel) {
      this.mdcFloatingLabel.destroy()
    }
  },
  methods: {
    onParentInit () {
      const label = this.getLabel()
      if (label instanceof MDCFloatingLabel) {
        if (this.mdcFloatingLabel instanceof MDCFloatingLabel) this.mdcFloatingLabel.destroy()
        this.mdcFloatingLabel = label
      }
    }
  }
}
</script>
