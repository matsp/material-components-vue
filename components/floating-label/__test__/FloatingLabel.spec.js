import 'mutationobserver-shim'
import { mount } from '@vue/test-utils'
import FloatingLabel from '../FloatingLabel.vue'

describe('FloatingLabel', () => {
  it('should mount', () => {
    let wrapper = mount(FloatingLabel)
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.vm.$data.mdcFloatingLabel).toBeDefined()
  })

  it('should render with no prop', () => {
    let wrapper = mount(FloatingLabel)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.classes()).toContain('mdc-floating-label')
  })

  it('should render with floatAbove', () => {
    let wrapper = mount(FloatingLabel, {
      propsData: {
        floatAbove: true
      }
    })
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.classes()).toContain('mdc-floating-label--float-above')
  })

  it('should render with shake', () => {
    let wrapper = mount(FloatingLabel, {
      propsData: {
        shake: true
      }
    })
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.classes()).toContain('mdc-floating-label--shake')
  })
})
