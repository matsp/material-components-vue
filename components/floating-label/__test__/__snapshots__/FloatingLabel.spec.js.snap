// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloatingLabel should render with floatAbove 1`] = `<label class="mdc-floating-label mdc-floating-label--float-above"></label>`;

exports[`FloatingLabel should render with no prop 1`] = `<label class="mdc-floating-label"></label>`;

exports[`FloatingLabel should render with shake 1`] = `<label class="mdc-floating-label mdc-floating-label--shake"></label>`;
